### PR TITLE
Fix #1137, exclude system databases

### LINKED ISSUE #1137: [MetaStore] Check if user is using a default database, e.g., mysql, test

### PR DESCRIPTION
**Pull request by PHILO-HE**
The databses "mysql", "sys", "information_schema", "INFORMATION_SCHEMA", "performance_schema" and "PERFORMANCE_SCHEMA" are excluded for SSM metastore.

### CODE REVIEW

File: smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java [hunk 1]
```diff
@@ -353,6 +353,23 @@ public static MetaStore getDBAdapter(
           LOG.warn("Database URL not specified, using " + purl);
         }
 
+        if (purl.startsWith(MetaStoreUtils.MYSQL_URL_PREFIX)) {
+          String dbName;
+          String[] dbNameNotAllowed = new String[]{
```
Comment (huafengw) [inline smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java]: This string array can be a final static variable.

File: smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java [hunk 2]
```diff
@@ -353,6 +353,23 @@ public static MetaStore getDBAdapter(
...
+                  "mysql", "sys", "information_schema", "INFORMATION_SCHEMA", "performance_schema", "PERFORMANCE_SCHEMA"
+          };
+          if (purl.contains("?")) {
+            dbName = purl.substring(purl.indexOf("/", 13) + 1, purl.indexOf("?"));
```
Comment (huafengw) [inline smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java]: I think this line of code is very risky. Is there any other proper way to do the extraction? Or any other util tool can recognize the pattern?

**Comment by huafengw** [comment]
And every time you want to create a pull request, please checkout a new branch from the latest trunk to avoid the duplicated commit history.

File: smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java (hunk 1 above)
Comment (PHILO-HE) [reply smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java]: Yes, I have modified. Thanks!

File: smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java (hunk 2 above)
Comment (PHILO-HE) [reply smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java]: A proper method is used to get the db name in the updated code. Thanks!

**Comment by huafengw** [comment]
LGTM +1

**Pull request merged**